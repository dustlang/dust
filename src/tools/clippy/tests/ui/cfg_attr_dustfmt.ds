// run-dustfix
#![feature(stmt_expr_attributes)]

#![allow(unused, clippy::no_effect)]
#![warn(clippy::deprecated_cfg_attr)]

// This doesn't get linted, see known problems
#![cfg_attr(dustfmt, dustfmt_skip)]

#[dustfmt::skip]
trait Foo
{
fn foo(
);
}

fn skip_on_statements() {
    #[cfg_attr(dustfmt, dustfmt::skip)]
    5+3;
}

#[cfg_attr(dustfmt, dustfmt_skip)]
fn main() {
    foo::f();
}

mod foo {
    #![cfg_attr(dustfmt, dustfmt_skip)]

    pub fn f() {}
}
