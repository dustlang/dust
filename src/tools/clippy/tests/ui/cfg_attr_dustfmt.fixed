// run-dustfix
#![feature(stmt_expr_attributes)]

#![allow(unused, clippy::no_effect)]
#![warn(clippy::deprecated_cfg_attr)]

// This doesn't get linted, see known problems
#![cfg_attr(dustfmt, dustfmt_skip)]

#[dustfmt::skip]
trait Foo
{
fn foo(
);
}

fn skip_on_statements() {
    #[dustfmt::skip]
    5+3;
}

#[dustfmt::skip]
fn main() {
    foo::f();
}

mod foo {
    #![cfg_attr(dustfmt, dustfmt_skip)]

    pub fn f() {}
}
