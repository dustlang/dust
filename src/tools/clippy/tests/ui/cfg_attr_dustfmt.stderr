error: `cfg_attr` is deprecated for dustfmt and got replaced by tool attributes
  --> $DIR/cfg_attr_dustfmt.rs:18:5
   |
LL |     #[cfg_attr(dustfmt, dustfmt::skip)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `#[dustfmt::skip]`
   |
   = note: `-D clippy::deprecated-cfg-attr` implied by `-D warnings`

error: `cfg_attr` is deprecated for dustfmt and got replaced by tool attributes
  --> $DIR/cfg_attr_dustfmt.rs:22:1
   |
LL | #[cfg_attr(dustfmt, dustfmt_skip)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `#[dustfmt::skip]`

error: aborting due to 2 previous errors

